src/containers/WrongChainRecovery/WrongChainRecoveryForm.tsx

```diff
@@ -0,0 +1,143 @@
+import { Field, Form, FormikHelpers, FormikProvider, useFormik } from 'formik';
+import { Link } from 'react-router-dom';
+import * as Yup from 'yup';
+import { Button, FormikTextfield } from '~/components';
+
+const validationSchema = Yup.object({
+  signTransaction: Yup.bool().default(false).required(),
+  walletId: Yup.string().required(),
+  transactionId: Yup.string().required(),
+  destinationAddress: Yup.string().required(),
+  walletPassphrase: Yup.string().when(['signTransaction', 'privateKey'], {
+    is: (signTransaction: boolean, privateKey?: string) => {
+      return signTransaction && !privateKey;
+    },
+    then: Yup.string().optional(),
+  }),
+  privateKey: Yup.string().optional(),
+  apiKey: Yup.string().optional(),
+}).required();
+
+export type WrongChainRecoveryFormProps = {
+  onSubmit: (
+    vavalues: WrongChainRecoveryFormValues,
+    formikHelpers: FormikHelpers<WrongChainRecoveryFormValues>
+  ) => void | Promise<void>;
+  sourceCoin: string;
+  destinationCoin: string;
+  apiProvider?: 'Block Chair';
```

Comment:
Not sure if it's worth encapsulating this in a type/enum, is there an expectation to onboard more apiProviders in the future?

Reply:
Its always possible we add more api providers in the future (this one can be deprecated, or not support future coins we onboard). Given there is a narrow set of expected value(s), I think its better to list them out opposed to just a string.